### PR TITLE
feat: add GrpcServer, GrpcClient, ConnectionPool skel

### LINKED ISSUE #53: [grpc] server & client skel
**Subject**
@zeroFruit 

**Detail**
Skeleton the grpc server, grpc client which would use `Connection` to communicate with other nodes

### PR DESCRIPTION
**Pull request by zeroFruit**
## Related Issue

resolve: #53 

## Description

Worked on skeletoning GrpcServer, GrpcClient, ConnectionPool

## Checklist

Thank you for your contribution.
Before submitting this PR, please make sure:

- [x] Applying goimports
- [ ] Test case
- [x] End of Work

### CODE REVIEW

File: comm.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+type GrpcServer struct {
+	connHandler ConnHandler
+	errHandler  ErrHandler
+	ip          string
+	lis         net.Listener
+	meta        map[string]string
+}
+
+func NewServer(meta map[string]string) *GrpcServer {
```
Comment (AgwaB) [inline comm.go]: what is `meta`?

File: conn.go [hunk 2]
```diff
@@ -72,3 +72,23 @@ func (conn *GrpcConnection) Start() error {
...
+
+type Broadcaster interface {
+	Broadcast(data []byte, typ string, succCallback func(interface{}), errCallback func(error))
+}
+
+type ConnectionPool struct {
+	connMap map[ConnID]Connection
+}
+
+func (p *ConnectionPool) Broadcast(data []byte, typ string, succCallback func(interface{}), errCallback func(error)) {
```
Comment (AgwaB) [inline conn.go]: `succCallback`is so ambiguous. I recommend just `successCallback`

File: conn.go [hunk 3]
```diff
@@ -72,3 +72,23 @@ func (conn *GrpcConnection) Start() error {
...
+	panic("implement me w/ test case :-)")
+}
+
+func (p *ConnectionPool) AddConn(id ConnID, conn Connection) {
```
Comment (AgwaB) [inline conn.go]: I think naming `Add` and `Remove` is enough

File: comm.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+func (s *GrpcServer) Listen(ip string) {
+	panic("implement me w/ test case :-)")
+}
+
+func (s *GrpcServer) Stop() {
+	panic("implement me w/ test case :-)")
+}
+
+type DialOpts struct {
```
Comment (AgwaB) [inline comm.go:44]: what is purpose of this `DialOpts` struct?

File: conn.go (hunk 2 above)
Comment (zeroFruit) [reply conn.go]: I'm planning to remove callback functions

File: comm.go (hunk 1 above)
Comment (zeroFruit) [reply comm.go]: meta data of our grpc server, but maybe we don't need it now

I'll remove it

File: comm.go (hunk 4 above)
Comment (zeroFruit) [reply comm.go:44]: with DialOpts we can set option for dial

In practical, one of DialOpts is dial timeout !

**Review by AgwaB — APPROVED** [review]

**Review by AgwaB — APPROVED** [review]

**Pull request merged**